game/world/objects/fireplace.cpp

```diff
@@ -1,9 +1,11 @@
+#include <phoenix/vobs/mob.hh>
+
 #include "fireplace.h"
 
-FirePlace::FirePlace(Vob* parent, World& world, ZenLoad::zCVobData& vob, Flags flags)
-  :Interactive(parent,world,vob,flags){
-  fireVobtreeName = std::move(vob.oCMobFire.fireVobtreeName);
-  fireSlot        = std::move(vob.oCMobFire.fireSlot);
+FirePlace::FirePlace(Vob* parent, World& world, const std::unique_ptr<phoenix::vob>& vob, Flags flags)
+  : Interactive(parent,world,vob,flags){
+  fireVobtreeName = ((const phoenix::vobs::mob_fire*) vob.get())->vob_tree;
```

Comment:
C-like cast is not desirable, so `reinterpret_cast`.
Also, probably it's better to handle cast on caller side and pass `phoenix::vobs::mob_fire&` here